### DISCUSSION TITLE
How to handle server sent events in Unpoly?

### DISCUSSION THREAD

**Discussion by formoxyz** [unpoly/unpoly#376] [Q&A] [answered]
Hi,
I am trying to understand if Unpoly supports server sent events?
For ex: htmx has this
```
<body hx-sse="connect:/news_updates">
    <div hx-trigger="sse:new_news" hx-get="/news"></div>
</body>
```

and Hotwire has [Turbo Streams](https://turbo.hotwired.dev/handbook/streams)

Is there any such equivalent in Unpoly? Please advise. Thanks.

**Comment by formoxyz** [comment]
I think [up-hungry] might be it. But I am confused how to use it with Server Sent Events. An example would be great. Thanks.

**Comment by formoxyz** [comment, marked as answer]
On further studying I think one option can be using javascript to listen to server sent events and then using up.render() to update the html received from the server.
```
es = new EventSource("/subscribe/imbox")
es.onmessage = function(event) {
  up.render({ target: '.two', fragment: event.data })
}
```

Please suggest. Thanks.

**Reply by adam12** [reply]
This is a fine approach. Another is to reload the element using `up.reload`.
